holidays/ical.py

```diff
@@ -0,0 +1,212 @@
+            holiday_name.replace("\\", "\\\\").replace(",", "\\,").replace(":", "\\:")
+        )
+        event_uid = f"{uuid.uuid4()}@{self.holidays_version}.holidays.local"
+
+        lines = [
+            "BEGIN:VEVENT",
+            f"DTSTAMP:{self.ical_timestamp}",
+            f"UID:{event_uid}",
+            self._fold_line(f"SUMMARY:{sanitized_holiday_name}"),
+            f"DTSTART;VALUE=DATE:{dt:%Y%m%d}",
+            f"DURATION:P{holiday_length}D",
+            "END:VEVENT",
+        ]
+
+        return lines
+
+    def generate(self, return_bytes: bool = False) -> Union[str, bytes]:
+        """
+        Generate iCalendar data.
+
+        :param return_bytes:
+            If True, return bytes instead of string.
+
+        :return:
+            The complete iCalendar data
+            (string or UTF-8 bytes depending on return_bytes).
+        """
+        lines = [
+            "BEGIN:VCALENDAR",
+            f"PRODID:-//Vacanza//Open World Holidays Framework v{self.holidays_version}//"
+            f"{self.language}",
+            "VERSION:2.0",
+            "CALSCALE:GREGORIAN",
+        ]
+
+        sorted_dates = sorted(self.holidays.keys())
+        # Merged continuous holiday with the same name and use `DURATION` instead.
+        i = 0
+        while i < len(sorted_dates):
+            dt = sorted_dates[i]
+            names = self.holidays.get_list(dt)
+
+            for name in names:
+                days = 1
+                while (
+                    i + days < len(sorted_dates)
+                    and sorted_dates[i + days] == sorted_dates[i] + timedelta(days=days)
+                    and name in self.holidays.get_list(sorted_dates[i + days])
+                ):
+                    days += 1
+
+                lines.extend(self._generate_event(dt, name, days))
+
+            i += days
+
+        lines.append("END:VCALENDAR")
+        lines.append("")
+
+        output = CONTENT_LINE_DELIMITER.join(lines)
+        return output.encode() if return_bytes else output
+
+    def export_ics(self, file_path: str) -> None:
+        """
+        Export the calendar data to a .ics file.
+
+        While RFC 5545 does not specifically forbid filenames for .ics files, but it’s advisable
+        to follow general filesystem conventions and avoid using problematic characters.
+
+        :param file_path:
+            Path to save the .ics file, including the filename (with extension).
+        """
+        # Generate and write out content (always in bytes for .ics)
+        content = self.generate(return_bytes=True)
+        if not content:
+            raise ValueError("Generated content is empty or invalid.")
+
+        try:
+            with open(file_path, "wb") as file:
+                file.write(content)  # type: ignore  # this is always bytes, ignoring mypy error.
+        except OSError as e:
+            raise OSError(f"An error occurred while writing the file '{file_path}': {str(e)}")
```

Comment:
```suggestion
        with open(file_path, "wb") as file:
            file.write(content)  # type: ignore  # this is always bytes, ignoring mypy error.
```
Why catch an exception just to append additional text to it? 😉

Reply:
Maybe I shouldn't have tried to solve all the issues at 1 AM lmao